dask/tests/test_base.py

```diff
@@ -28,81 +37,79 @@ def f3(a):
-    assert normalize_function(curry(f2)) != normalize_function(curry(f1))
-    assert (normalize_function(curry(f2, b=1)) ==
-            normalize_function(curry(f2, b=1)))
-    assert (normalize_function(curry(f2, b=1)) !=
-            normalize_function(curry(f2, b=2)))
+    assert normalize_function(tz.curry(f2)) == normalize_function(tz.curry(f2))
+    assert normalize_function(tz.curry(f2)) != normalize_function(tz.curry(f1))
+    assert (normalize_function(tz.curry(f2, b=1)) ==
+            normalize_function(tz.curry(f2, b=1)))
+    assert (normalize_function(tz.curry(f2, b=1)) !=
+            normalize_function(tz.curry(f2, b=2)))
 
 
 def test_tokenize():
     a = (1, 2, 3)
     assert isinstance(tokenize(a), (str, bytes))
 
 
+@pytest.mark.skipif('not np')
 def test_tokenize_numpy_array_consistent_on_values():
-    np = pytest.importorskip('numpy')
     assert tokenize(np.random.RandomState(1234).random_sample(1000)) == \
            tokenize(np.random.RandomState(1234).random_sample(1000))
 
 
+@pytest.mark.skipif('not np')
 def test_tokenize_numpy_array_supports_uneven_sizes():
-    np = pytest.importorskip('numpy')
     tokenize(np.random.random(7).astype(dtype='i2'))
 
 
+@pytest.mark.skipif('not np')
 def test_tokenize_discontiguous_numpy_array():
-    np = pytest.importorskip('numpy')
     tokenize(np.random.random(8)[::2])
 
 
+@pytest.mark.skipif('not np')
 def test_tokenize_numpy_datetime():
-    np = pytest.importorskip('numpy')
     tokenize(np.array(['2000-01-01T12:00:00'], dtype='M8[ns]'))
 
 
+@pytest.mark.skipif('not np')
 def test_tokenize_numpy_scalar():
-    np = pytest.importorskip('numpy')
     assert tokenize(np.array(1.0, dtype='f8')) == tokenize(np.array(1.0, dtype='f8'))
     assert (tokenize(np.array([(1, 2)], dtype=[('a', 'i4'), ('b', 'i8')])[0])
          == tokenize(np.array([(1, 2)], dtype=[('a', 'i4'), ('b', 'i8')])[0]))
 
 
+@pytest.mark.skipif('not np')
 def test_tokenize_numpy_array_on_object_dtype():
-    np = pytest.importorskip('numpy')
     assert tokenize(np.array(['a', 'aa', 'aaa'], dtype=object)) == \
            tokenize(np.array(['a', 'aa', 'aaa'], dtype=object))
     assert tokenize(np.array(['a', None, 'aaa'], dtype=object)) == \
            tokenize(np.array(['a', None, 'aaa'], dtype=object))
     assert tokenize(np.array([(1, 'a'), (1, None), (1, 'aaa')], dtype=object)) == \
            tokenize(np.array([(1, 'a'), (1, None), (1, 'aaa')], dtype=object))
     if sys.version_info[0] == 2:
-        assert tokenize(np.array([unicode("Rebeca Alón",encoding="utf-8")], dtype=object)) == \
-               tokenize(np.array([unicode("Rebeca Alón",encoding="utf-8")], dtype=object))
-
+        assert tokenize(np.array([unicode("Rebeca Alón", encoding="utf-8")], dtype=object)) == \
+               tokenize(np.array([unicode("Rebeca Alón", encoding="utf-8")], dtype=object))
 
 
+@pytest.mark.skipif('not np')
 def test_tokenize_numpy_memmap():
-    np = pytest.importorskip('numpy')
```

Comment:
There were several spots where `importorskip` was used incorrectly. Many tests also relied on several optional packages. When fixing the former I also just cleaned up and standardized all the other instances. Not necessarily better, but more uniform.